### ISSUE TITLE
Chart.display() issue with highly-specific VSCode extensions in Ubuntu

### ISSUE THREAD

**Issue by hcpchris** [highcharts-for-python/highcharts-core#46]
A user on [StackOverflow](https://stackoverflow.com/questions/76096200/highcharts-for-python-doesnt-render-in-jupyter/76101379) reported a bug when attempting to generate the basic line tutorial from the demo when running Jupyter Notebooks on Ubuntu, in VSCode, with the following VSCode extensions enabled: Jupyter, Jupyter Cell Tags, Jupyter Notebook Renderers, Jupyter Slide Show, Jupyter Keymap. After upgrading to v.1.1.1 and disabling/re-enabling the extensions, the problem was resolved.

The question to investigate is whether there is a reproducible conflict with one or more of those extensions, and to determine the specific combination of extensions and environment that can lead to errors, and to see if there's a solution to the problem.

**Comment by romanb1337** [comment]
Hi! I have the same problem. Highcharts does not work in Jupiter Notebook, VS Code (Jupyter) and even Binder (from https://stock-docs.highchartspython.com/en/latest/demos.html). But works in Jupiter Lab
<img width="1348" alt="jupyternotebook" src="https://github.com/highcharts-for-python/highcharts-core/assets/128704794/b2f861cc-f7b4-421d-a01a-76c7b4973f68">
![vscode](https://github.com/highcharts-for-python/highcharts-core/assets/128704794/cd5f7b6d-a109-4b84-a979-a3217e3125d5)
<img width="1357" alt="binder" src="https://github.com/highcharts-for-python/highcharts-core/assets/128704794/6ab7db30-8f05-4335-bf54-46b78adf121c">
<img width="1339" alt="jupyterlab" src="https://github.com/highcharts-for-python/highcharts-core/assets/128704794/6b86ef2d-f5e1-4cef-ab59-3aed7ce7a765">

**Comment by hcpchris** [comment]
Thanks, @romanb1337 ! Sorry you're running into this issue in Jupyter Notebooks in VS Code. A couple questions and one suggestion:

**QUESTIONS**: 
1. Do you have any Jupyter Notebook-related extensions installed / running in your VS Code (e.g. Jupyter Cell Tags, Jupyter Slide Show, etc.)? If so, what are those extensions? We're trying to recreate this problem and it seems to be related to various extensions creating some sort of strange conflict, but have not yet been able to recreate the behavior. So knowing what extensions you are using (and ideally their version numbers) would prove super helpful.
2. What version of Highcharts Core for Python are you using? I assume v.1.1.1 (the latest release), but do want to double check.

**SUGGESTION**: 
Another customer who encountered this issue found that disabling the extensions, restarting VS Code, and then re-enabling their extensions, somehow "magically" solved the problem for them. That might be a reasonable workaround while we try to recreate / diagnose / resolve the underlying issue.

**Comment by romanb1337** [comment]
@hcpchris, yes i am using version 1.1.1 (macbook arm, macos 13.4*22F66, latest brew & conda, python 3.11.3 env https://gist.github.com/romanb1337/a17df1bb5b81b63b9250f13e307f3ec7 if it matters)

I tried switching extensions in all code, but it didn't help, here is a list of ext.:

ms-toolsai.jupyter
ms-toolsai.jupyter-keymap
ms-toolsai.jupyter-renderers
ms-toolsai.vscode-jupyter-cell-tags
ms-toolsai.vscode-jupyter-powertoys
ms-toolsai.vscode-jupyter-slideshow

<img width="294" alt="Снимок экрана 2023-05-19 в 20 02 34" src="https://github.com/highcharts-for-python/highcharts-core/assets/128704794/0610ba9e-2c00-4a2f-a8d1-e8559c9a4cd5">

**Comment by romanb1337** [comment]
@hcpchris i found this extension for VS Code, but Jupyter Notebook (browser ver) still not working

https://github.com/highcharts-for-python/highcharts-core/assets/128704794/df14ebdb-35fd-4e95-939a-4134d170803f

**Comment by hcpchris** [comment]
Thanks, @romanb1337 ! That's helpful that it seems to be an issue with the Jupyter Notebook Renderer extension. That'll help narrow down my experimentation. 

In the meantime, when you say that Jupyter Notebook "(browser ver)" is not working, does that mean that when you open a standard Jupyter Notebook within the browser (i.e. not within VS Code and *not* within Jupyter Lab, just running a standard Jupyter Notebook in the browser), the chart is *also* not rendering?

**Comment by romanb1337** [comment]
@hcpchris yes, when i open a standard Jupyter Notebook within the browser (i.e. not within VS Code and not within Jupyter Lab, just running a standard Jupyter Notebook in the browser), the chart is also not rendering

**Comment by hcpchris** [comment]
Perfect - thanks for confirming, @romanb1337 .

**Comment by yangrq1018** [comment]
Still not working in jupyter notebook (Sep 16). Lab is fine

**Comment by hcpchris** [comment]
Just as an FYI, at least *some* of the situations where this problem appears seems to be related to #66, where Jupyter Notebooks in certain contexts make use of RequireJS, which in turn produces odd (and oddly inconsistent) behavior in certain contexts / situations.

There's a new PR #112 that resolves #66, which should make it into the next minor release (**v.1.4.0**), which is slated to be released the week of October 2nd. I'm not certain whether resolving #66 will resolve *all* of the situations that give rise to this behavior in VSCode - but it should at least help with some of them. Even after the release, I'll be leaving this issue open to see the degree to which the problem has been resolved or to see if it still reoccurs. 

But here's hoping that fixing the RequireJS problem solves these inconsistent and hard-to-reproduce issues!

**Comment by hcpchris** [comment]
So with today's release of **v.1.4.0**, the bugfix for #66 was included. This should address the root cause for *some* of the VSCode / extension conflict issues. I don't believe it will resolve *all* of them, but it should hopefully resolve at least *some* of them. If folks who were encountering this issue could report back as to whether the new version resolves the issue for them, I'd be very grateful as it will help us determine additional root cause(s) of this weird behavior in VSCode.
